cpp/include/cuspatial/experimental/detail/ranges/multilinestring_range.cuh

```diff
@@ -23,15 +23,44 @@
 
 #include <cuspatial/cuda_utils.hpp>
 #include <cuspatial/detail/iterator.hpp>
-#include <cuspatial/experimental/geometry_collection/multipoint_ref.cuh>
+#include <cuspatial/experimental/geometry_collection/multilinestring_ref.cuh>
 #include <cuspatial/traits.hpp>
 #include <cuspatial/vec_2d.hpp>
 
 #include <iterator>
 
 namespace cuspatial {
 
-using namespace cuspatial::detail;
+using namespace detail;
+
+template <typename GeometryIterator, typename PartIterator, typename VecIterator>
+struct to_multilinestring_functor {
+  using difference_type = typename thrust::iterator_difference<GeometryIterator>::type;
+  GeometryIterator _geometry_begin;
+  PartIterator _part_begin;
+  VecIterator _point_begin;
+  VecIterator _point_end;
+
+  CUSPATIAL_HOST_DEVICE
+  to_multilinestring_functor(GeometryIterator geometry_begin,
+                             PartIterator part_begin,
+                             VecIterator point_begin,
+                             VecIterator point_end)
+    : _geometry_begin(geometry_begin),
+      _part_begin(part_begin),
+      _point_begin(point_begin),
+      _point_end(point_end)
+  {
+  }
+
+  CUSPATIAL_HOST_DEVICE auto operator()(difference_type i)
+  {
+    return multilinestring_ref{_part_begin + _geometry_begin[i],
+                               thrust::next(_part_begin + _geometry_begin[i + 1]),
+                               _point_begin,
+                               _point_end};
+  }
```

Comment:
Curious what you're using this for. A constructor that returns a slice of a `multilinestring_range`?

Reply:
Which is a multilinestring.